src/pm/hydra2/mpiexec/mpiexec_params.c

```diff
@@ -847,6 +847,25 @@ static HYD_status tree_width_fn(char *arg, char ***argv)
     goto fn_exit;
 }
 
+/* --pmi-args is an internal option used for singleton mode implementation (MPI-2)
+ * format of option is:
+ *   mpiexec --pmi-args <singleton_port> default_interface default_key <singleton_pid>
+ */
```

Comment:
A 5-part command line argument is uncommon and potentially breaks other parts of code (potentially future code). For example, since the rest of the args are all `-name arg` two-part, it is easy to skip unknown ones by assuming such convention. A 5-part argument will prevent that. There are two options: you could either break the four parts each with a `--name`, or simply forcing the use of quotes to make it a single string arg and internally break it up.